SoftServeCarMobile/src/activity/messages/chat/simple-message/SimpleMessage.tsx

```diff
@@ -1,94 +1,95 @@
+import "reflect-metadata";
+import { container } from 'tsyringe';
+import { SearchBar } from "react-native-elements";
 import SimpleMessageStyle from "./SimpleMessageStyle";
-import * as navigation from "../../../../components/navigation/Navigation";
 
-const SimpleMessage = (props: {
-    navigation: { navigate: (arg0: string) => void };
-}) => {
+
+const SimpleMessage = (props: any) => {
     const chatService = container.resolve(ChatService);
-    const [data, setData] = useState([]);
-    const [isLoading, setLoading] = useState(true);
+    const [filteredDataSource, setFilteredDataSource] = useState([]);
+    const [masterDataSource, setMasterDataSource] = useState([]);
+    const [search, setSearch] = useState('');
     const { user } = useContext(AuthContext);
 
     useEffect(() => {
-        chatService.getChat(user?.id).then((res) => {
-            const chats = res.data;
-            console.log(chats);
-            setData(chats);
-            setLoading(false);
-        });
+        chatService.getChat(user?.id)
+            .then(res => {
+                const chats = res.data;
+                console.log(chats);
+                setFilteredDataSource(chats);
+                setMasterDataSource(chats);
+            })
     }, []);
 
-    if (isLoading) {
+    const searchFilterFunction = (text) => {
+        if (text) {
+            const newData = masterDataSource.filter(function (item) {
+                const itemData = item.name
+                    ? item.name.toUpperCase()
+                    : ''.toUpperCase();
+                const textData = text.toUpperCase();
+                return itemData.indexOf(textData) > -1;
+            });
+            setFilteredDataSource(newData);
+            setSearch(text);
+        } else {
+            setFilteredDataSource(masterDataSource);
+            setSearch(text);
+        }
+    };
+
+    const renderHeader = () => {
+        return (props.isOpenFilter) ? (
+            <SearchBar
+                searchIcon={{ color: "black", size: 28 }}
+                onChangeText={(text) => searchFilterFunction(text)}
+                onClear={() => searchFilterFunction('')}
+                placeholder={'Search in Messages'}
+                value={search}
+                containerStyle={SimpleMessageStyle.containerStyle}
+                inputContainerStyle={SimpleMessageStyle.inputContainerStyle}
+            />) : <View />
+    };
+
+    const ItemView = ({ item }) => {
         return (
-            <View>
-                <ActivityIndicator size="large" color="#5500dc" />
+            <View style={SimpleMessageStyle.main}>
+                <View style={SimpleMessageStyle.button}>
+                    <View style={{ flexDirection: 'row' }}>
+                        <Image style={SimpleMessageStyle.image}
+                            source={{ uri: 'https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__340.jpg' }} />
```

Comment:
hardcoded resource, please move uri to const

Reply:
It just a temporary URL for better visibility, here I will use a component from notifications.